web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -86,74 +86,99 @@ CREATE OR REPLACE PROCEDURE {{.output_schema}}.column_check(SOURCE_SCHEMA VARCHA
   AS
   $$
 
-  var delim = '~';
-  var sourceColumns = list_cols_with_type(SOURCE_SCHEMA,SOURCE_TABLE,delim).split(delim);
-  var targetColumns = list_cols_with_type(TARGET_SCHEMA,TARGET_TABLE,delim).split(delim);
-
-  if (targetColumns.some(notIncludedIn(sourceColumns)) === true) {
-
-      throw "ERROR: Source table is missing column(s) which exist in target table.";
-
-  } else {
-
-      var columnAdditions = sourceColumns.filter(notIncludedIn(targetColumns));
-
-      if (sourceColumns.some(notIncludedIn(targetColumns)) === true) {
-          if ( AUTOMIGRATE !== 'TRUE' ) {
-              throw "ERROR: Target table is missing column(s),but automigrate is not enabled.";
-          } else {
-              // enforce order
-              for (var i = 0; i < targetColumns.length; i++) {
-                  if (targetColumns[i] !== sourceColumns[i]) {
-                      throw "ERROR: Can only migrate extra columns of the end of source"
-                  }
-              }
-              add_columns_to(TARGET_SCHEMA, TARGET_TABLE, columnAdditions.join(','));
-              return "ok.Columns added."
-          }
-      } else {
-          return "ok. Columns match."
-      }
+  column_check_stmt = `
+    WITH source_data AS (
+      SELECT
+        column_name,
+        ordinal_position,
+        character_maximum_length,
+        data_type,
+        CASE
+          WHEN isc.data_type='TEXT' THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
+          WHEN isc.data_type='NUMBER' THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
+          else CONCAT(isc.column_name, ' ', isc.data_type)
+        END AS column_definition
+      FROM information_schema.columns isc
+      WHERE table_schema = UPPER(:1)
+      AND table_name = UPPER(:2)
+      ORDER BY ordinal_position
+      ),
+
+  target_data AS (
+    SELECT
+      column_name,
+      ordinal_position,
+      character_maximum_length,
+      data_type
+    FROM information_schema.columns isc
+    where table_schema = UPPER(:3)
+    AND table_name = UPPER(:4)
+    ORDER BY ordinal_position
+    ),
+
+  varchar_check AS (
```

Comment:
I am wondering if we simply take the script from [here](https://github.com/snowplow/data-models/blob/ba0d08c27b9f226bf467aaedeb64d9f7a54df128/mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql#L117-L162)

And add after here:
https://github.com/snowplow/data-models/blob/ba0d08c27b9f226bf467aaedeb64d9f7a54df128/mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql#L156

```SQL
LISTAGG(CASE WHEN tc.column_name IS NOT NULL AND sc.character_maximum_length > tc.character_maximum_length THEN sc.column_name END, ',') as cols_w_incompatible_char_limits
```

Does that get us the details you need?

Reply:
Yes, it looks like it does. As agreed, to keep it in synch with the mobile model I have changed it according to the above.